aiohttp/web_response.py

```diff
@@ -341,8 +343,8 @@ async def _start_compression(self, request: "BaseRequest") -> None:
             # Encoding comparisons should be case-insensitive
             # https://www.rfc-editor.org/rfc/rfc9110#section-8.4.1
             accept_encoding = request.headers.get(hdrs.ACCEPT_ENCODING, "").lower()
-            for coding in ContentCoding:
-                if coding.value in accept_encoding:
+            for value, coding in CONTENT_CODINGS.items():
+                if value in accept_encoding:
```

Comment:
I'm struggling with this one. It looks to me like all we've saved is one attribute access...

With the dict, I was expecting a reversal of logic or something here, like `for enc in accept_encoding: CONTENT_CODINGS.get(enc)`.

Reply:
Iterating enums in python is surprisingly expensive:

We save:
- creating the iterator for ContentCoding enum
- creating a enum singleton for each one in the loop
- accessing the value

... maybe we can save more here but I was thinking the `in` 3x was cheaper than a `dict` lookup

Reply:
a dict get might make sense if we parsed the string but parsing is probably more expensive `Accept-Encoding: deflate, gzip;q=1.0, *;q=0.5`